1-theory/1-apis-services-microservices.adoc

```diff
@@ -0,0 +1,144 @@
+= APIs, Services, and Microservices
+
+Before we get too stuck into things we should probably get some terminology down so we arent just 
+throwing around these terms interchangeably like the majority of developers.
+
+== API
+
+The acronym API stands for “Application Program Interface”, and
+that’s what it says on the tin: an interface for interacting with an application.
+This is a rather generic term in the world of computer science and programming, as a
+Linux command-line utility would consider its command names and options to be an
+API, whilst a Java library would also consider the method names and arguments to be
+an API.
+
+Essentially it’s all the same concept, but when talking about Web APIs we’re talking
+about the utilization of network protocols such as HTTP, AMQP, etc., and instead of
+command-line arguments or method parameters, we’re working with URLs, query string
+parameters, or chunks of data (often JSON).
+
+The goal of a Web API is to provide other applications with access to specific subset of
+functionality and data that this application owns. Some APIs are public (anyone can get
+the weather from api.weather.gov), some are private (it’s probably tough to get access
+to JPMorgan Chase’s Enterprise Customers API), and some are a mixture.
+
+APIs can give you data for a single-page JavaScript application, handle payment
+information so you don’t have to worry about storing credit card details, post a
+Facebook status on a users timeline, or share the same data across a myriad of different
+devices. Watch that Netflix show on your Xbox, it’s all good, APIs have you covered.
+
+APIs are built by all sorts of folks; maybe another company, some large like Google or
+Facebook, startups, governments, or charity organizations. They could also be
+coworkers on another team, or another department, on another floor (or continent).
+
+== Service
+
+A service is another very overloaded term in computer science, and programming in genral. A lot 
+of developers are used to MVC (Model, View, Controller) and a service is just "any code that didnt
+fit into a model, view, or controller". 
+
+When folks talk about services in API-land, they are usually talking about "Service-orientated Architecture".
```

Comment:
```suggestion
When folks talk about services in API-land, they are usually talking about "Service-oriented Architecture".
```

Orientated or oriented? Wikipedia wins? Since we're being picky about names in this chapter I feel fully validated to point out this boring inconsistency (and another one on line 44) :D